igor-web/src/main/java/com/netflix/spinnaker/igor/artifacts/ArtifactService.java

```diff
@@ -33,9 +33,26 @@
 
   /** Used to populate the manual trigger dropdown with options */
   @Nonnull
-  List<String> getArtifactVersions(@Nonnull String name, String releaseStatus);
+  List<String> getArtifactVersions(
+      @Nonnull String type, @Nonnull String name, String releaseStatus);
 
   /** Used to fetch a specific artifact for decorating a trigger */
   @Nonnull
-  Artifact getArtifact(@Nonnull String name, @Nonnull String version);
+  Artifact getArtifact(@Nonnull String type, @Nonnull String name, @Nonnull String version);
+
+  /**
+   * Used to populate the manual trigger dropdown with options (backwards-compatibility for Debian)
```

Comment:
Why do you need these backwards compatibility functions when you default the parameter in the controller?

Reply:
Good point. I started from the bottom up, so I added these here in case there were any other users beyond that controller, but I think there are none, so I can probably remove them safely.